frontend/mr-admin-flate/src/components/modal/SlettAvtaleGjennomforingModal.tsx

```diff
@@ -0,0 +1,127 @@
+import { Button, Heading, Modal } from "@navikt/ds-react";
+import {
+  ApiError,
+  Avtale,
+  Opphav,
+  Tiltaksgjennomforing,
+} from "mulighetsrommet-api-client";
+import { useEffect } from "react";
+import classNames from "classnames";
+import { useNavigate } from "react-router-dom";
+import { XMarkOctagonFillIcon } from "@navikt/aksel-icons";
+import styles from "../modal/Modal.module.scss";
+import { Lenkeknapp } from "../lenkeknapp/Lenkeknapp";
+import { UseMutationResult } from "@tanstack/react-query";
+
+interface SlettTiltaksgjennomforingModalprops {
+  modalOpen: boolean;
+  handleCancel: () => void;
+  data: Tiltaksgjennomforing | Avtale;
+  mutation: UseMutationResult<string, unknown, string>;
+  dataType: "tiltaksgjennomforing" | "avtale";
+}
+
+const SlettAvtaleGjennomforingModal = ({
+  modalOpen,
+  handleCancel,
+  data,
+  mutation,
+  dataType,
+}: SlettTiltaksgjennomforingModalprops) => {
+  const navigate = useNavigate();
+
+  const fraArena = data?.opphav === Opphav.ARENA;
+
+  useEffect(() => {
+    if (mutation.isSuccess) {
+      navigate(`/${dataType}er`);
+      return;
+    }
+  }, [mutation]);
+
+  const handleDelete = () => {
+    mutation.mutate(data.id);
+  };
+
+  function headerInnhold() {
+    return (
+      <div className={styles.heading}>
+        <XMarkOctagonFillIcon className={styles.warningicon} />
+
+        {fraArena ? (
+          <span>
+            {dataType === "tiltaksgjennomforing"
+              ? "Gjennomføringen"
+              : "Avtalen"}{" "}
```

Comment:
Du kan vurdere å lage et objekt som inneholder tekster for hhv. gjennomføring og avtaler, så slipper du all denne ternary-logikken alle steder.

Feks.

```ts
const tekster = {gjennomforing: {navnPlural: "Gjennomføringen"}, avtale: {navnPlural: "Avtalen"}}

Så kan du bruke det slik: 
tekster[dataType].navnPlural istedenfor å sjekke datatypen overalt. Det vil gjøre koden mer ryddig tror jeg.

```